src/commonMain/kotlin/com/charleskorn/kaml/YamlConfiguration.kt

```diff
@@ -83,12 +85,35 @@ public enum class SingleLineStringStyle {
     DoubleQuoted,
     SingleQuoted,
     Plain,
-    ;
+
+    /**
+     * This is the same as [SingleLineStringStyle.Plain], except strings that are not strictly strings in YAML
+     * without quotes will be escaped with the escape style defined in [AmbiguousEscapeStyle].
+     *
+     * True -> Boolean -> Escaped
+     *
+     * 0xAB -> Hexadecimal -> Escaped
+     *
+     * 1 -> Int -> Escaped
+     *
+     * 1.2 -> Float -> Escaped
+     *
+     * 1.2.3 -> Unambiguously a string -> Not escaped
+     *
+     * @since 0.51.0
```

Comment:
I think this could be a little clearer:

```suggestion
     * This is the same as [SingleLineStringStyle.Plain], except strings that could be misinterpreted as other types
     * will be quoted with the escape style defined in [AmbiguousEscapeStyle].
     *
     * For example, the strings "True", "0xAB", "1" and "1.2" would all be quoted,
     * while "1.2.3" and "abc" would not be quoted.
```